saga/sagaState.go

```diff
@@ -6,30 +6,40 @@ import (
 )
 
 /*
+ * Additional data about tasks to be committed to the log
+ * This is Opaque to SagaState, but useful data to persist for
+ * results or debugging
+ */
+type taskData struct {
```

Comment:
There are a lot of maps from taskID to some kind of data. Would it be worth making a taskState struct that stores all of them (and is also immutable)?